### PR TITLE
Properly encode URLs (and refactor api call logic)

### PR DESCRIPTION
**Pull request by larskarlitski**
I only wanted to ensure that URLs are encoded properly, but I accidentally refactored quite a bit of the API call functions. I think it is more consistent now and reads much better.

Please see individual commits.

### CODE REVIEW

File: core/apiCalls.js [hunk 1]
```diff
@@ -108,18 +108,7 @@ export function commitToWorkspaceApi(blueprint) {
...
-      .apiFetch("/api/v0/blueprints/diff/" + blueprintId + "/NEWEST/WORKSPACE")
-      .then(data => {
-        resolve(data);
-      })
-      .catch(e => {
-        console.log("Error fetching diff", e);
-        reject();
-      });
-  });
-  return p;
```
Comment (KKoukiou) [inline core/apiCalls.js]: Just for clarity, here they were calling reject inside the catch, so there was not an issue of not propagating the exception up. But I agree that it's useless and well removed.

File: data/BlueprintApi.js [hunk 2]
```diff
@@ -1,88 +1,48 @@
...
-          console.log(`Error committing blueprint: ${e}`);
-          NotificationsApi.closeNotification(undefined, "committing");
-          NotificationsApi.displayNotification(blueprint.name, "commitFailed");
-          reject();
-        });
-    });
-    return p;
-  }
-
-  postedBlueprintData(blueprint) {
```
Comment (jgiardino) [inline data/BlueprintApi.js]: This is still being used in sagas/blueprints.js, (e.g. `BlueprintApi.postedBlueprintData`).

**Comment by jgiardino** [comment]
These updates are great! Thanks for making them. I only had one comment above so far but will continue reviewing later today.

File: data/BlueprintApi.js (hunk 2 above)
Comment (larskarlitski) [reply data/BlueprintApi.js]: Oops. Thanks for the catch. I've uploaded a fix.

**Pull request merged**